daml-foundations/daml-ghc/daml-stdlib-src/DA/Internal/Template.daml

```diff
@@ -87,12 +87,28 @@ class Template c where
     internalArchive = magic @"archive"
 
 
+-- Deliberately not exported.
+data Consuming = PreConsuming  -- Archive before executing exercise body.
+               | PostConsuming -- Execute exercise body then archive.
+               | NonConsuming  -- Don't archive.
+
+-- | HIDE
+nonconsuming : NoEvent c e -> Consuming
+nonconsuming _ = NonConsuming
+
+preconsuming : NoEvent c e -> Consuming
```

Comment:
Add HIDE on this and postconsuming please

Reply:
✔️